api/handlers_secrets.go

```diff
@@ -91,15 +116,96 @@ func (s *server) SecretCreateHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *server) SecretShowHandler(w http.ResponseWriter, r *http.Request) {
+	w = LogWriter{w}
+	vars := mux.Vars(r)
+	account := vars["account"]
+	smService, ok := s.smServices[account]
+	if !ok {
+		msg := fmt.Sprintf("secretsmanager service not found for account: %s", account)
+		handleError(w, apierror.New(apierror.ErrNotFound, msg, nil))
+		return
+	}
+	id := vars["secret"]
+	secret, err := smService.GetSecretMetaDataWithFilter(r.Context(), id, func(out *secretsmanager.DescribeSecretOutput) bool {
+		log.Debugf("checking tags for %s to be sure it's part of the org %s", aws.StringValue(out.Name), s.org)
+		for _, tag := range out.Tags {
+			if aws.StringValue(tag.Key) == "spinup:org" && aws.StringValue(tag.Value) == s.org {
+				log.Debugf("%s has matching org tag and is part of the %s org, adding to the list", aws.StringValue(out.Name), s.org)
+				return true
+			}
+		}
+		return false
+	})
+	if err != nil {
+		handleError(w, errors.Wrap(err, "unable to get secret from the secretsmanager service"))
+		return
+	}
+
+	j, err := json.Marshal(secret)
+	if err != nil {
+		handleError(w, errors.Wrap(err, "unable to marshal response from the secretsmanager service"))
+		return
+	}
+
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(http.StatusOK)
-	w.Write([]byte{})
+	w.Write(j)
 }
 
 func (s *server) SecretDeleteHandler(w http.ResponseWriter, r *http.Request) {
+	w = LogWriter{w}
+	vars := mux.Vars(r)
+	account := vars["account"]
+	smService, ok := s.smServices[account]
+	if !ok {
+		msg := fmt.Sprintf("secretsmanager service not found for account: %s", account)
+		handleError(w, apierror.New(apierror.ErrNotFound, msg, nil))
+		return
+	}
+
+	id := vars["secret"]
+	window := int64(7)
+	q := r.URL.Query()
+	if w := q.Get("window"); w != "" {
+		i, err := strconv.ParseInt(w, 10, 64)
+		if err != nil {
+			log.Warnf("failed to parse window size as integer: %s", err)
```

Comment:
Shouldn't we return an error in this case?

Reply:
👍